src/cantools/logreader.py

```diff
@@ -32,269 +42,244 @@ def __init__(self, channel: str,
-        data = match_object.group('can_data')
-        data = data.replace(' ', '')
-        data = binascii.unhexlify(data)
+    def parse_timestamp(self, match_object: 're.Match[str]') -> 'tuple[TimestampType, TimestampFormat]':
         timestamp = datetime.datetime.strptime(match_object.group('timestamp'), "%Y-%m-%d %H:%M:%S.%f")
         timestamp_format = TimestampFormat.ABSOLUTE
-
-        return DataFrame(channel=channel, frame_id=frame_id, data=data, timestamp=timestamp, timestamp_format=timestamp_format)
+        return timestamp, timestamp_format
 
 
 class PCANTracePatternV10(BasePattern):
     """
     Reference for PCAN trace patterns: https://www.peak-system.com/produktcd/Pdf/English/PEAK_CAN_TRC_File_Format.pdf
     1) 1841 0001 8 00 00 00 00 00 00 00 00
+
+    >>> PCANTracePatternV10().match(" 1) 1841 0001 8 00 00 00 00 00 00 00 00") #doctest: +ELLIPSIS
+    <logreader.DataFrame object at ...>
     """
     pattern = re.compile(
-        r'^\s*?\d+\)\s*?(?P<timestamp>\d+)\s+(?P<can_id>[0-9A-F]+)\s+(?P<dlc>[0-9])\s+(?P<can_data>[0-9A-F ]*)$')
+        r'^\s*?\d+\)\s*?(?P<timestamp>\d+)\s+(?P<can_id>[0-9A-F]+)\s+(?P<dlc>[0-9])\s+(?P<can_data>RTR|[0-9A-F ]*)$')
 
-    @staticmethod
-    def unpack(match_object):
-        """
-        >>> PCANTracePatternV10().match(" 1) 1841 0001 8 00 00 00 00 00 00 00 00") #doctest: +ELLIPSIS
-        <logreader.DataFrame object at ...>
-        """
-        channel = 'pcanx'
+    def unpack(self, match_object: 're.Match[str]') -> 'DataFrame|None':
+        channel = self.parse_channel(match_object)
         frame_id = int(match_object.group('can_id'), 16)
-        data = match_object.group('can_data')
-        data = data.replace(' ', '')
-        data = binascii.unhexlify(data)
+        is_extended_frame = len(match_object.group('can_id')) > 4
+        data, is_remote_frame = self.parse_data(match_object)
         millis = float(match_object.group('timestamp'))
         # timestamp = datetime.datetime.strptime(match_object.group('timestamp'), "%Y-%m-%d %H:%M:%S.%f")
         timestamp = datetime.timedelta(milliseconds=millis)
         timestamp_format = TimestampFormat.RELATIVE
 
-        return DataFrame(channel=channel, frame_id=frame_id, data=data, timestamp=timestamp, timestamp_format=timestamp_format)
+        return DataFrame(channel=channel, frame_id=frame_id, is_extended_frame=is_extended_frame, data=data, is_remote_frame=is_remote_frame, timestamp=timestamp, timestamp_format=timestamp_format)
 
+    def parse_channel(self, match_object: 're.Match[str]') -> str:
+        return 'pcanx'
 
-class PCANTracePatternV11(BasePattern):
+    def parse_data(self, match_object: 're.Match[str]') -> 'tuple[bytes, bool]':
+        data = match_object.group('can_data')
+        if data == 'RTR':
+            is_remote_frame = True
+            data = bytes(0)
+        else:
+            is_remote_frame = False
+            data = data.replace(' ', '')
+            data = binascii.unhexlify(data)
```

Comment:
probably this can be replaced by
```suggestion
            data = bytes.fromhex(data)
```